src/webview/components/options-widget.tsx

```diff
@@ -297,33 +302,32 @@ export class OptionsWidget extends React.Component<OptionsWidgetProps, {}> {
     protected doHandleAdvancedOptionsDropdownChange(event: DropdownChangeEvent): void {
         const id = event.target.id as InputId;
         const value = event.target.value;
-
         switch (id) {
             case InputId.WordsPerGroup:
-                this.updateConfiguration({
-                    id: 'groupings.wordsPerGroup',
-                    value: Number(value),
-                });
+                this.props.updateRenderOptions({ wordsPerGroup: Number(value) });
                 break;
             case InputId.GroupsPerRow:
-                this.updateConfiguration({
-                    id: 'groupings.groupsPerRow',
-                    value: Number(value),
-                });
+                this.props.updateRenderOptions({ groupsPerRow: Number(value) });
                 break;
             default: {
                 throw new Error(`${id} can not be handled. Did you call the correct method?`);
             }
         }
     }
 
-    protected updateConfiguration(
-        viewConfigurationChangeRequest: MemoryDisplayConfigurationChangeRequest
-    ): void {
-        return messenger.sendNotification(
-            setMemoryDisplayConfigurationType,
-            HOST_EXTENSION,
-            viewConfigurationChangeRequest
-        );
+    protected handleColumnActivationChange: (labelSelected: string, newSelectionState: boolean) => void = (label, state) => this.doHandleColumnActivationChange(label, state);
+    doHandleColumnActivationChange(label: string, state: boolean): void {
```

Comment:
Maybe `shouldRender` rather than `state` to avoid overloading the term `state` in a React context.